### ISSUE TITLE
Define ansible role per VM

### ISSUE THREAD

**Issue by nvgoldin** [lago-project/lago#429]
When https://github.com/lago-project/lago/pull/428 gets in, it will deduce the VM ansible role from the vm-type, this might be good for ovirtlago, but would enforce creating a VM plugin per role. I suggest to add to the VM metadata a 'role' parameter, for example:
```yaml
dom1:
....
    metadata:
        role: httpd
```
The ``role`` would default to the VM type if not defined.
I'm familiar enough with ansible requirements, maybe it should be 'ansible' specific? for instance:
```yaml
dom1:
....
    metadata:
    ....
      ansible:
         role: 
         ...
         (other ansible metadata params)
        ...
```
some relevant past issue/work done by  @ehelms : https://github.com/lago-project/lago/issues/281

**Comment by ifireball** [comment]
"role" is something very specific in the Ansible world - A set of tasks and related things that are wrapped together. A role in Ansible is somewhat like a class and a module in Puppet wrapped together into one.

I suggest we find a more generic way to mark up and group VMs, maybe 'labels' or 'tags'.

**Comment by nvgoldin** [comment]
And then parse ``lable`` -> ``role`` in ansible specific commands? Our 'metadata' is pretty context free at the moment. My question is exactly that - will we need more 'ansible' specific paramaters there, then we might bundle them all together.

**Comment by ifireball** [comment]
An Ansible playbook is the way you assign roles to hosts or groups thereof (its more then that, b/c it also specifies the order in which things happen, what can happen in parallel, etc.). This is a core concept of Ansible and Lago shouldn't mess with that too much.

In the end Lago should not be doing a lot more then providing an Inventory file to Ansible. An Inventory file can:
* Define hosts and how to connect to them
* Group hosts
* Define host metadata (AKA variables. At the individual host level or at the group level)

All the above can be useful in the Ansible contest but also in other contexts. consider for example:
* `lago start <name of group of VMs>`
* Making an equvalent of the Ansible commands but for Puppet e.g. generate puppet ENC data where variables become parameter assignments

Bottom line - I thing we should avoid Ansible-specific terminology in the init file and instead come up with concepts that can have wider use.

**Comment by mykaul** [comment]
I'd be happy with Ansible specific terminology. It's a widely-known, used and agreed-upon terminology.

**Comment by ifireball** [comment]
Well we can use 'groups' to group hosts together and 'variables' to assign arbitrary host properties. That is the same terminology Ansible uses, while still using generic terms  and without explicitly spelling-out 'Ansible', which is redundant in my view.
